source/common/config/cds_json.cc

```diff
@@ -117,8 +117,9 @@ void CdsJson::translateCluster(const Json::Object& json_cluster,
   } else {
     ASSERT(string_type == "sds");
     cluster.set_type(envoy::api::v2::Cluster::EDS);
-    Utility::sdsConfigToEdsConfig(sds_config.value(), *cluster.mutable_eds_config());
-    cluster.mutable_deprecated_v1()->set_service_name(json_cluster.getString("service_name", ""));
+    auto* eds_cluster_config = cluster.mutable_eds_cluster_config();
+    Utility::sdsConfigToEdsConfig(sds_config.value(), *eds_cluster_config->mutable_eds_config());
+    eds_cluster_config->set_service_name(json_cluster.getString("service_name", ""));
```

Comment:
nit: if you default this to name() if "service_name" does not exist, you can get rid of the ternary operator below and just always use service_name() I think?